File: Prog3 - Delivery/DisqueRango - cpp/DisqueRango/disquerango.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
#include <algorithm>

#include "entregador.h"
#include "cardapio.h"
#include "carregador.h"
#include "iterador.h"
#include "cliente.h"
#include "pedidos.h"
#include "interface.h"
#include "relatorio.h"
#include "tempo.h"

using namespace std;


bool comparaCodigo(const cardapio c1, const cardapio c2)
{
  return (c1.codigo < c2.codigo);
}

int main(int argc,char** argv){
	tempo t;
	vector<entregador> ent;
	vector<cardapio> card;
	vector<cliente> clientes;
	vector<pedidos> pedidosDia;
	carregador c(argc,argv,ent,card,t);
	interface i(clientes,card,pedidosDia,ent,argv[1],t);
	relatorio r(ent,card,pedidosDia);
}

File: Prog3 - Delivery/DisqueRango/src/DisqueRango/DisqueRango.java
package DisqueRango;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Scanner;


public class DisqueRango {	
	public static void main(String[] args){
		Scanner s = new Scanner(System.in);
		Calendar can = Calendar.getInstance();
		ArrayList<Entregador> entregador = new ArrayList<Entregador>();
		ArrayList<Cardapio> cardapio = new ArrayList<Cardapio>();
		ArrayList<Cliente> clientes = new ArrayList<Cliente>();
		ArrayList<Pedidos> pedidosDia = new ArrayList<Pedidos>();
		new Carregador(args,cardapio,entregador,clientes,can);
		new Interface(clientes,cardapio,pedidosDia,entregador,s,can,args[0]);
		new Persistencia().Serializador(cardapio, entregador, clientes);
		new Relatorio(entregador,cardapio,pedidosDia,s,can);
		s.close();
	}
}
